File: PRACTICA1/RUBY/Deepspace/BetaPowerEfficientSpaceStation.rb
#encoding:utf-8
require './BetaPowerEfficientSpaceStationToUI.rb'
module Deepspace


    # @author Lucía Salamanca López, Celia Arias Martínez
class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
    @@EXTRAEFFICIENCY=1.2  #float, atributo de instancia de la clase

    def initialize(estacion)  #SpaceStation
        super(estacion)
    end 

    def fire   #devuelve un float
        if (Dice.new().extraEfficiency)
            return super*@@EXTRAEFFICIENCY
        else
            return super 
        end
    end

    def self.extraefficiency
        @@EXTRAEFFICIENCY
    end

    def getUIversion
        BetaPowerEfficientSpaceStationToUI.new(self)
    end

    def  to_s 
        out = "Estación espacial eficiente beta: "
        out+= super
    end
end 
end

File: PRACTICA1/JAVA/deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Lucía Salamanca López
 *         Celia Arias Martínez
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
   
    private static final float EXTRAEFFICIENCY=1.2f;
    
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public float fire(){
        Dice dado = new Dice();
        if(dado.extraEfficiency())
            return super.fire()*EXTRAEFFICIENCY;
        else 
            return super.fire();
    }
    
    @Override
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
         return new BetaPowerEfficientSpaceStationToUI(this);
    }
}
